core/src/cost_update_service.rs

```diff
@@ -166,6 +169,19 @@ impl CostUpdateService {
                     );
                     }
                 }
+
+                let updated_estimated_program_cost =
+                    cost_model.read().unwrap().find_instruction_cost(program_id);
```

Comment:
could we return this from the `upsert_instruction_cost()` cost above instead of grabbing the lock again

Reply:
The `upsert` only adds data, in order to get `cost` it needs to run a calculation, therefore `upsert` and `find_cost` are two separated functions. But with Trent's suggestion above to use MAX_UNITS as yard stick, no need to call to find cost again here.